pytket/binders/passes.cpp

```diff
@@ -28,34 +28,27 @@ using json = nlohmann::json;
 
 namespace tket {
 
-void update_routing_config(RoutingConfig &config, py::kwargs kwargs) {
-  if (kwargs.contains("swap_lookahead"))
-    config.depth_limit = py::cast<unsigned>(kwargs["swap_lookahead"]);
-  if (kwargs.contains("bridge_lookahead"))
-    config.distrib_limit = py::cast<unsigned>(kwargs["bridge_lookahead"]);
-  if (kwargs.contains("bridge_interactions"))
-    config.interactions_limit =
-        py::cast<unsigned>(kwargs["bridge_interactions"]);
-  if (kwargs.contains("bridge_exponent"))
-    config.distrib_exponent = py::cast<unsigned>(kwargs["bridge_exponent"]);
-}
 static PassPtr gen_cx_mapping_pass_kwargs(
     const Architecture &arc, const PlacementPtr &placer, py::kwargs kwargs) {
-  RoutingConfig config = {};
-  update_routing_config(config, kwargs);
+  RoutingMethodPtr method = std::make_shared<LexiRouteRoutingMethod>(100);
+  std::vector<RoutingMethodPtr> config = {method};
+  if (kwargs.contains("config")) {
+    config = py::cast<std::vector<RoutingMethodPtr>>(kwargs["config"]);
+  }
```

Comment:
It's a weird one - probably ideally yes, however in that case we're defining an empty RoutingMethodPtr object and waiting to assign to it depending on the configuration (so that the name persists in object for passing to gen_cx_mapping_pass). Given LexiRouteRoutingMethod has nearly no overhead in creation, I'd rather we create it and then remove it if there is another option, than leaving the pointer (or opening up some if / else logic).